Bottom: a62fdc32be85e4e3c18ebadd8d5ec0483ea4dd9a
Top:    f8f1619e85a9834d725362a8870da7dfc3b4a9b7
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-30 00:47:23 +0100

Refresh of major-re-write-to-drawing-rout

---

diff --git a/src/draw.h b/src/draw.h
index 6cb124e..f6dc31f 100644
--- a/src/draw.h
+++ b/src/draw.h
@@ -85,7 +85,7 @@ void DrawPlainVia (PinTypePtr, bool);
 void DrawRegularText (LayerTypePtr, TextTypePtr);
 void DrawEMark (ElementTypePtr, Coord, Coord, bool);
 void DrawHole (PinTypePtr);
-void DrawRats (BoxType *);
+void DrawRats (const BoxType *);
 /* TEMPORARY */
 
 #endif
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index fdad579..4c62846 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1492,7 +1492,7 @@ clearPad_callback_solid (const BoxType * b, void *cl)
 }
 
 static void
-DrawMask (int side, BoxType * screen)
+GhidDrawMask (int side, BoxType * screen)
 {
   int thin = TEST_FLAG(THINDRAWFLAG, PCB) || TEST_FLAG(THINDRAWPOLYFLAG, PCB);
 
@@ -1523,7 +1523,7 @@ DrawMask (int side, BoxType * screen)
 }
 
 static int
-DrawLayerGroup (int group, const BoxType * screen)
+GhidDrawLayerGroup (int group, const BoxType * screen)
 {
   int i, rv = 1;
   int layernum;
@@ -1777,14 +1777,14 @@ ghid_draw_everything (BoxTypePtr drawn_area)
     if (!global_view_2d &&
         gui->set_layer (SWAP_IDENT ? "componentmask" : "soldermask",
                         SWAP_IDENT ? SL (MASK, TOP) : SL (MASK, BOTTOM), 0)) {
-        DrawMask (side, drawn_area);
+        GhidDrawMask (side, drawn_area);
         gui->end_layer ();
       }
   }
 
   /* draw all layers in layerstack order */
   for (i = ngroups - 1; i >= 0; i--) {
-    DrawLayerGroup (drawn_groups [i], drawn_area);
+    GhidDrawLayerGroup (drawn_groups [i], drawn_area);
 
 #if 1
     if (!global_view_2d && i > 0 &&
@@ -1834,7 +1834,7 @@ ghid_draw_everything (BoxTypePtr drawn_area)
   /* Draw the solder mask if turned on */
   if (gui->set_layer (SWAP_IDENT ? "soldermask" : "componentmask",
                       SWAP_IDENT ? SL (MASK, BOTTOM) : SL (MASK, TOP), 0)) {
-    DrawMask (side, drawn_area);
+    GhidDrawMask (side, drawn_area);
     gui->end_layer ();
   }
